spring-kafka/src/main/java/org/springframework/kafka/retrytopic/BackOffPolicyProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.retrytopic;
+
+import org.springframework.retry.backoff.BackOffPolicy;
+import org.springframework.retry.backoff.ExponentialBackOffPolicy;
+import org.springframework.retry.backoff.ExponentialRandomBackOffPolicy;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.backoff.UniformRandomBackOffPolicy;
+
+/**
+ * Provides a {@link BackOffPolicy} from the given parameters.
+ * Original logic from Spring Retry project.
+ *
+ * @author Tomaz Fernandes
+ * @since 2.8.4
+ */
+class BackOffPolicyProvider {
```

Comment:
Why do we need this extra class, when there is only one method which could migrate to that `RetryTopicConfigurationBuilder` ?